apps/sla/time.py

```diff
@@ -23,6 +23,74 @@ def add_days(dt: datetime, days: int) -> datetime:
 EASTERN = timezone("US/Eastern")
 UTC = timezone("UTC")
 WEEKEND = (6, 7)  # Sat, Sun
+HOLIDAY_DEADLINE_MONTH = 1  # January
+HOLIDAY_DEADLINE_DAY = 8
+
+
+def make_sure_date_is_date_object(day):
+    """
+    Make sure the given day is a date object.
+    """
+    try:
+        return day.date()  # get date from datetime
+    except AttributeError:
+        pass  # is already a date
+    return day
+
+
+def go_to_next_holiday_deadline(day):
+    """
+    Go to the very deadline (January 8th) after the given day.
+    Preserves the original time and timezone of the input datetime.
+    """
+    # Preserve original time and timezone if it's a datetime
+    if isinstance(day, datetime):
+        orig_tz = day.tzinfo
+        orig_time = (day.hour, day.minute, day.second, day.microsecond)
+    else:
+        orig_tz = None
+        orig_time = (0, 0, 0, 0)
```

Comment:
I'm not 100% sure if we want to keep the timezone, looking at the current code for weekends, the timezones are converted to eastern. @osoukup can probably shine some light on this.